### PR TITLE
Allow jobs to parameterize how many VCPUs are allocated

### PR DESCRIPTION
**Pull request by ttung**
### Description
This may make the jobs a taaaaaaaaaaad faster.

### Test plan
[Test in my sandbox](https://us-west-2.console.aws.amazon.com/states/home?region=us-west-2#/executions/details/arn:aws:states:us-west-2:473004499091:execution:aspen-rdev-ttung-nextstrain-swipe-spot-sfn:gisaid-more-vcpu)

### CODE REVIEW

**Review by shanaxel42 — APPROVED** [review]
approved!

**Comment by jgadling** [comment]
I'm a little curious at how this impacts provisioning. Based on the instance types available for R5d, it looks like AWS is allocating 8 vcpus per 64GB of memory, so will this just be a no-op?

**Comment by ttung** [comment]
I believe that it allocates you the cores you request.  So as it is, you would get a 1 core 64 GB container.

**Comment by jgadling** [comment]
Alright, I looked this up: https://docs.aws.amazon.com/batch/latest/userguide/job_definition_parameters.html

```
vcpus
    The number of vCPUs reserved for the container. This parameter maps to CpuShares in the Create a container
section of the Docker Remote API and the --cpu-shares option to docker run. Each vCPU is equivalent to 1,024 CPU shares
```

TL;DR: I don't think this will hurt anything, but I don't think it'll help either.

[It's a longer story](https://en.wikipedia.org/wiki/Completely_Fair_Scheduler), but this essentially sets a weight for each container running on the system. Since we've set the weight to the same value for all jobs, this means that if all containers on the system are requesting CPU time at the same time, they'll do an even split of CPU resources.

**Comment by ttung** [comment]
Our tree builds are vcpu=4, the gisaid ingest is vcpu=32, and everything else is vcpu=1.

**Comment by ttung** [comment]
But you're right, this is a different issue!  Turns out upstream ncov is playing with compression, which has huge perf implications.
